### PR TITLE
sumo/gcc 12.1 upgrade

### PR DESCRIPTION
**Pull request by jhershbe**
Upgrade gcc and also all the components that are needed to compile with gcc 12.1

Along with gcc, we need to upgrade binutils, elfutils, valgrind, & gdb to be compatible with the new dwarf 5 format [1].

Versions needed from [2]:
GDB 8.0, or higher
valgrind 3.17.0
elfutils 0.172, or higher (for use with systemtap, dwarves/pahole, perf and libabigail)
dwz 0.14

[1] - https://gcc.gnu.org/gcc-11/changes.html
[2] - https://git.openembedded.org/openembedded-core/commit/?id=d07d4d739ae17787017f771dd2068fda0e836722

Other package updates are to allow clean builds with the updated compiler.

Patches were cherry-picked from OE master.

### CODE REVIEW

**Comment by amstewart** [comment]
Could you add some more detail to the PR description:
* Why are we making this change in the sumo mainline?
* What recipes are you intentionally upgrading here?
* What upstream OE-core ref did these patches come from? `master`?
* What testing has been done on this patchset? I'm expecting at least...
    * That the [`packagegroup-ni-base`](https://github.com/ni/meta-nilrt/blob/715829f461d449f3926819e01e932cd993aab926/recipes-core/packagegroups/packagegroup-ni-base.bb) bitbake target builds without errors.
    * That the [`packagegroup-ni-desirable`](https://github.com/ni/meta-nilrt/blob/715829f461d449f3926819e01e932cd993aab926/recipes-core/packagegroups/packagegroup-ni-desirable.bb) bitbake target builds without errors.
    * That the `niconsole-image`, `niconsole-image-safe`, and `niconsole-dkms-image` [targets](https://github.com/ni/meta-nilrt/tree/715829f461d449f3926819e01e932cd993aab926/recipes-core/images) build without

File: meta/recipes-kernel/lttng/babeltrace2_2.0.3.bb
Comment (amstewart) [inline meta/recipes-kernel/lttng/babeltrace2_2.0.3.bb]: This commit needs a signoff and an explanation for why we're reverting the target commit.

File: meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb
Comment (amstewart) [inline meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb]: Again; why the revert?

File: meta/recipes-kernel/perf/perf.bb
Comment (amstewart) [inline meta/recipes-kernel/perf/perf.bb:84]: I need some more commentary here about why it is appropriate to ignore these warnings. Is this fixed in the `perf` upstream?

**Review by amstewart — CHANGES_REQUESTED** [review]

File: meta/recipes-connectivity/openssh/openssh_8.3p1.bb
Comment (shruthi-ravi) [inline meta/recipes-connectivity/openssh/openssh_8.3p1.bb:11]: Can you add why you dropped the dependency on virtual/crypt in 9d81605f7dc3df4b83f3a4ae2d07d0a622518920 and 1609b06f1586b2e392ceb17b209fe281888e4b0c?

Comment (jhershbe) [reply meta/recipes-connectivity/openssh/openssh_8.3p1.bb:11]: These are patches that add a dependency on a virtual component that doesn't exist in sumo, so there is nothing there to depend on. An artifact of upgrading some components but not all.

File: meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb
Comment (jhershbe) [reply meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb]: I think it was already coming in from a later release

File: meta/recipes-kernel/perf/perf.bb
Comment (jhershbe) [reply meta/recipes-kernel/perf/perf.bb:84]: Basically we are stuck on an old kernel, so instead of pulling in fixes to the kernel for this, we just ignore the warnings like the previous compiler did.

File: meta/recipes-kernel/lttng/babeltrace2_2.0.3.bb
Comment (jhershbe) [reply meta/recipes-kernel/lttng/babeltrace2_2.0.3.bb]: removed

File: meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb
Comment (amstewart) [reply meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb]: In sumo we use the dhcp packages to manage networking, I think. Can you confirm that the `.patch` is no longer needed, and give that justification in your commit message?

https://github.com/ni/openembedded-core/pull/68/commits/6917854e7db2551696454fb670b8be3f44adb5d7

File: meta/recipes-kernel/perf/perf.bb
Comment (amstewart) [reply meta/recipes-kernel/perf/perf.bb:84]: https://github.com/ni/openembedded-core/pull/68/commits/75611681599427339562980bb974c02ba1ca62f9

Could you add that justification to the commit message and sign-off the commit?

**Review by amstewart — CHANGES_REQUESTED** [review]
I'm not going to block the PR on this, but if you're already re-cherry-picking the commits in this set, could you add the cherry-pick trailer (`-x`) and your signoff (`-s`) when you do?

File: meta/recipes-kernel/perf/perf.bb
Comment (jhershbe) [reply meta/recipes-kernel/perf/perf.bb:84]: Done.

File: meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb
Comment (jhershbe) [reply meta/recipes-connectivity/dhcp/dhcp_4.4.1.bb]: I updated the commit to reflect that we revert it so that the upstream version will apply cleanly.

a311013e1fd0e7b874dfe11bb3cdf4839cd91c9f

**Review by amstewart — APPROVED** [review]
OK. I think this PR is good enough to go into the academic mainline.

(For posterity) We don't intend on merging this into the NILRT ARM32 (sumo) mainline, as that branch has a different support policy.

**Pull request merged**
